addon/mixins/version-header-handler.js

```diff
@@ -18,6 +18,18 @@ const {
   Mixin,
 } = Ember;
 
+function fetchHeader(header, headers) {
```

Comment:
So this means we have an O(N) operation for EVERY request after this change. Is this going to be a problem?

Reply:
Mmm...it's probably fine but I think I have a better idea...